engine/collector_instance.go

```diff
@@ -61,12 +71,19 @@ func collect(dbp zesty.DBProvider) error {
 					}
 				} else {
 					// run found resolution
-					logrus.WithFields(logrus.Fields{"resolution_id": r.PublicID}).Debugf("Instance Collector: collected crashed resolution %s", r.PublicID)
+					log.WithFields(logrus.Fields{"resolution_id": r.PublicID}).Debugf("collected crashed resolution %s", r.PublicID)
 					_ = GetEngine().Resolve(r.PublicID)
+
+					// waiting between two resolve, so others instances can also select tasks
+					time.Sleep(2 * time.Second)
+					resolved++
 				}
 			}
 			// no resolutions left to retry, delete instance
-			i.Delete(dbp)
+			if remaining, err := getRemainingResolution(dbp, i); err == nil && remaining == 0 {
+				log.Debugf("collected all resolution from %d, deleting instance from instance list", i.ID)
```

Comment:
I think the log level should be higher. Info?
It should be enabled on a typical prod deployment, to retrace what happened